libcaf/libcaf/repository.py

```diff
@@ -691,6 +701,231 @@ def create_tag(self, tag_name: str, commit_hash: str, author: str, message: str)
     @requires_repo
     def status(self) -> Sequence[Diff]:
         return self.diff(self.head_commit(), self.working_dir)
+    
+    def _resolve_checkout_target(self, target: str) -> Ref | None:
+        """ Resolve a checkout target to a Ref.
+
+            The resolution is done by scanning the refs directory so we can distinguish between
+            branches and tags by location.
+            If there is a collision between a branch and a tag name, the branch takes precedence similair to Git.
+
+            :param target: The checkout target provided by the user.
+            :return: The resolved Ref object or None if the target does not exist.
+        """
+
+        if not isinstance(target, str):
+            msg = f'Invalid checkout target type: {type(target)}'
+            raise RefError(msg)
+
+        if len(target) == HASH_LENGTH and all(c in HASH_CHARSET for c in target):
```

Comment:
HashRef was created so that code doesn't have to check for this, instead it can just check if it's holding a HashRef